src/null-safety/faq.md

```diff
@@ -0,0 +1,292 @@
+    migration to null safety.
+-   Or, mark the field `late final`. This enforces that it's initialized exactly
+    once. It must be initialized before it can be read.
+
+## How should I migrate a `built_value` class?
+
+Getters that were annotated `@nullable` should instead have nullable types; then
+remove all `@nullable` annotations. For example:
+
+```dart
+@nullable
+int get count;
+```
+
+becomes
+
+```dart
+int? get count; //  Variable initialized with ?
+```
+
+Getters that were *not* marked `@nullable` should *not* have nullable types,
+even if the migration tool suggests them. Add `!` hints as needed then rerun the
+analysis.
+
+## How should I migrate a factory that can return `null`?
+
+_Prefer factories that do not return null._ We have seen code that meant to
+throw an exception due to invalid input but instead ended up returning null.
+
+Instead of:
+
+{:.bad}
+{% prettify dart tag=pre+code %}
+  factory StreamReader(dynamic data) {
+    StreamReader reader;
+    if (data is ByteData) {
+      reader = BlockReader(data);
+    } else if (data is Map) {
+      reader = JSONBlockReader(data);
+    }
+    return reader;
+  }
+{% endprettify %}
+
+Do:
+
+{:.good}
+{% prettify dart tag=pre+code %}
+  factory StreamReader(dynamic data) {
+    if (data is ByteData) {
+      // Move the readIndex forward for the binary reader.
+      return BlockReader(data);
+    } else if (data is Map) {
+      return JSONBlockReader(data);
+    } else {
+      throw ArgumentError('Unexpected type for data');
+    }
+  }
+{% endprettify %}
+
+
+If the intent of the factory was indeed to return null, then you can turn it
+into a static method so it is allowed to return `null`.
+
+## How should I migrate an `assert(x != null)` that now shows as unnecessary?
+
+The assert will be unnecessary when everything is fully migrated, but for now it
+*is* needed if you actually want to keep the check. Options:
+
+-   Decide that the assert is not really necessary, and remove it. This is a
+    change in behavior when asserts are enabled.
+-   Decide that the assert can be checked always, and turn it into
+    `ArgumentError.checkNotNull`. This is a change in behavior when asserts are
+    not enabled.
+-   Keep the behavior exactly as is: add `// ignore:
+    unnecessary_null_comparison` to bypass the warning.
+
+## How should I migrate a runtime null check that now shows as unnecessary?
+
+An explicit runtime null check, for example `if (arg == null) throw
+ArgumentError(...)`, will be flagged as an unnecessary comparison if you make
+`arg` non-nullable.
+
+But, the check *is* still needed when running in mixed mode. Until everything is
+fully migrated and the code switches to running in sound mode, it will be
+possible for `arg` to be null.
+
+The simplest way to preserve behavior is change the check into
+[`ArgumentError.checkNotNull`](https://api.dart.dev/stable/2.9.3/dart-core/ArgumentError/checkNotNull.html).
```

Comment:
In general, you should delete version numbers from URLs, so they automatically get the latest stable (or whatever) channel version of the doc.

```suggestion
[`ArgumentError.checkNotNull`](https://api.dart.dev/stable/dart-core/ArgumentError/checkNotNull.html).
```